src/Support/Field.php

```diff
@@ -165,11 +192,22 @@ protected function getResolver(): ?Closure
         };
     }
 
-    private function instanciateSelectFields(array $arguments, int $depth = null): SelectFields
+    /**
+     * @param array<int,mixed> $arguments
+     * @param int $depth
+     * @param array<string,mixed> $fieldsAndArguments
+     * @return SelectFields
+     */
+    private function instanciateSelectFields(array $arguments, int $depth = null, array $fieldsAndArguments): SelectFields
```

Comment:
I would move around the args, feels strange to have an optional parameter right in the middle (i.e. I suggest to make it the last one)